umbrella/router.go

```diff
@@ -102,16 +101,11 @@ func getFuncRoutingInfo(reqUrlRoot string, values url.Values, cmdReg *map[string
 	}
 
 	funcName := argsList[0]
-	confirmedFuncName, err := confirmFunctionIsRegistered(funcName, funcReg)
+	funcRoutingInfo, err := confirmFunctionIsRegistered(funcName, funcReg)
 	if err != nil {
 		return nil, err
 	}
 
-	funcRoutingInfo, funcRoutingInfoOk := funcReg[confirmedFuncName]
```

Comment:
A Product Manager will be upset here because you just erased the message for when the function is not registered. How do we distinguish between whether there was an error on the way to finding the function routing info and whether it was registered at all?

Reply:
Ah -- I wasn't sure how to handle that, since I returned the same message in the function if nothing was found.

Reply:
If there's no potential for error on the way to finding the function info, then it's fine to move this one error into your function. However, are you sure there can be no other issues on the way to finding the routing info?

Reply:
Definitely not sure, haha -- I am trying to wrap my head around what exactly was lost, though (as well as the errors `go build` is now revealing to me that I'm working through).

Previously, the assignment was returning the key's value and  a boolean about whether or not a key existed into `funcRoutingInfoOk`, and returning a message if it was false. Trying to reason about this, I think I'm currently doing the same thing (in concept if not execution): return the function info and nil in place of an error if it exists, or don't return the function info, and return the same message that was there previously if `funcRoutingInfoOk` was false as `err`. Or in short, I'm struggling to see what exactly was lost. 

I think if I changed `err` to just be a boolean, and returned the the fmt.Errorf message when it was false, it would be nearly identical in an error reporting sense, wouldn't it?

That confusion aside, go build has shown me some other problems that I have to figure out -- Go is bullying me even in this small task, haha. So this might even become moot as I figure those out.